pxr/imaging/plugin/hdRpr/volume.cpp

```diff
@@ -348,20 +393,40 @@ void HdRprVolume::Sync(
 				return;
 			}
 
-
-			/*
 			voxelSize = baseGrid->voxelSize();
 
-			if (baseGrid->isType<openvdb::BoolGrid>())			fillGridOnWithDefaultColor< openvdb::BoolGrid::Ptr, openvdb::BoolGrid::ValueOnIter >(openvdb::gridPtrCast<openvdb::BoolGrid>(baseGrid), grid, idx, gridOnBBSize);
-			else if (baseGrid->isType<openvdb::FloatGrid>())	fillGridOnWithDefaultColor< openvdb::FloatGrid::Ptr, openvdb::FloatGrid::ValueOnIter >(openvdb::gridPtrCast<openvdb::FloatGrid>(baseGrid), grid, idx, gridOnBBSize);
-			else if (baseGrid->isType<openvdb::DoubleGrid>())	fillGridOnWithDefaultColor< openvdb::DoubleGrid::Ptr, openvdb::DoubleGrid::ValueOnIter >(openvdb::gridPtrCast<openvdb::DoubleGrid>(baseGrid), grid, idx, gridOnBBSize);
-			else if (baseGrid->isType<openvdb::Int32Grid>())	fillGridOnWithDefaultColor< openvdb::Int32Grid::Ptr, openvdb::Int32Grid::ValueOnIter >(openvdb::gridPtrCast<openvdb::Int32Grid>(baseGrid), grid, idx, gridOnBBSize);
-			else if (baseGrid->isType<openvdb::Int64Grid>())	fillGridOnWithDefaultColor< openvdb::Int64Grid::Ptr, openvdb::Int64Grid::ValueOnIter >(openvdb::gridPtrCast<openvdb::Int64Grid>(baseGrid), grid, idx, gridOnBBSize);
-			else if (baseGrid->isType<openvdb::Vec3IGrid>())	fillGridOnWithDefaultColor< openvdb::Vec3IGrid::Ptr, openvdb::Vec3IGrid::ValueOnIter >(openvdb::gridPtrCast<openvdb::Vec3IGrid>(baseGrid), grid, idx, gridOnBBSize);
-			else if (baseGrid->isType<openvdb::Vec3SGrid>())	fillGridOnWithDefaultColor< openvdb::Vec3SGrid::Ptr, openvdb::Vec3SGrid::ValueOnIter >(openvdb::gridPtrCast<openvdb::Vec3SGrid>(baseGrid), grid, idx, gridOnBBSize);
-			else if (baseGrid->isType<openvdb::Vec3DGrid>())	fillGridOnWithDefaultColor< openvdb::Vec3DGrid::Ptr, openvdb::Vec3DGrid::ValueOnIter >(openvdb::gridPtrCast<openvdb::Vec3DGrid>(baseGrid), grid, idx, gridOnBBSize);
-			else if (baseGrid->isType<openvdb::StringGrid>())	fillGridOnWithDefaultColor< openvdb::StringGrid::Ptr, openvdb::StringGrid::ValueOnIter >(openvdb::gridPtrCast<openvdb::StringGrid>(baseGrid), grid, idx, gridOnBBSize);
-			*/
+			if (baseGrid->isType<openvdb::BoolGrid>())
+				voxelSize = voxelSize;
+				//fillGridOnWithDefaultColor< openvdb::BoolGrid::Ptr, openvdb::BoolGrid::ValueOnIter >(openvdb::gridPtrCast<openvdb::BoolGrid>(baseGrid), grid, idx, gridOnBBSize);
+			else if (baseGrid->isType<openvdb::FloatGrid>())
```

Comment:
Would this make more sense to do 
if (baseGrid->isType<openvdb::FloatGrid>()) {
    ... 
} else {
    voxelSize = voxelSize;
}

Unless the intention is to fill in more here?

Reply:
code refactored in latest commit